apps/src/templates/rubrics/LearningGoals.jsx

```diff
@@ -352,6 +363,9 @@ export default function LearningGoals({
                 setLoaded(teacherFeedbacksLoaded.current[index]);
                 setDisplayUnderstanding(understandingLevels.current[index]);
               }
+              if (index === learningGoals.length - 1) {
+                setDoneLoading(true);
+              }
```

Comment:
I like this solution to the race condition you were experiencing. I actually had to use something similar for the product tour to make sure the rubric is fully loaded before the tour starts.